qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -217,6 +218,24 @@ def _add_cal_metadata(self, experiment_data: ExperimentData):
         """
         pass
 
+    def _transpiled_circuits(self) -> List[QuantumCircuit]:
+        """Override the transpiled circuits method to bring in the inst_map.
+
+        The calibrated schedules are transpiled into the circuits using the instruction
+        schedule map which can be provided through the options. However, instances of
+        :class:`InstructionScheduleMap` are not serializable. This is fine since they
+        do not need to be serialized. Only instances of :class:`Calibrations` need to
+        be serialized. Here, we add the instruction schedule map to the transpile options
+        for transpilation and then remove it to avoid serialization issues.
+
+        Returns:
+            A list of transpiled circuits.
+        """
+        self.set_transpile_options(inst_map=self.calibrations.default_inst_map)
```

Comment:
I think this approach is really clean. On the other hand if #701 can move finalize call before the transpile, this doesn't need to override transpile method just to set an option. The important thing here is not to pass heavy instmap to json serializer, but this can be done by bypassing `set_*` method. What do you think @chriseclectic

Reply:
I think overriding transpiled circuits is the correct approach because this is controlling how the transpiled circuits are generated. But I wouldn't recommend doing it by temporarily modifying class attributes as is currently done, see my above comment for suggestion that does the same thing but without temporarily changing the internal state.